src/hooks/useFetcher/index.ts

```
@@ -4,13 +4,19 @@ import { useRef, useState } from 'react'
 
 type Success<T> = [T] extends [never] ? Record<string, never> : { json: T }
 
+// Conditional return type based on whether T is provided
+type FetcherReturn<T> = [T] extends [never] ? void : T
```

Biome (1.9.4)

[error] void is confusing outside a return type or a type parameter.

Unsafe fix: Use undefined instead.


(lint/suspicious/noConfusingVoidType)